util/stopwaiter/stopwaiter_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package stopwaiter
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/util/testhelpers"
+)
+
+const testStopDelayWarningTimeout = 350 * time.Millisecond
+
+type TestStruct struct{}
+
+func TestStopWaiterStopAndWaitTimeoutShouldWarn(t *testing.T) {
+	logHandler := testhelpers.InitTestLog(t, log.LvlTrace)
+	sw := StopWaiter{}
+	testCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	sw.Start(context.Background(), &TestStruct{})
+	sw.LaunchThread(func(ctx context.Context) {
+		<-testCtx.Done()
+	})
+	go sw.stopAndWaitImpl(testStopDelayWarningTimeout)
```

Comment:
stopAndWaitImpl is blocking, so I start it in separate go routine, though I am not sure if that's the best solution